### PR TITLE
Fix replicator task time-out error

### PR DESCRIPTION
**Pull request by priyanka25081999**
- Apply the solution to fix the s3replicator task time-out error.
- Update add_job function and also add debug logs.

Signed-off-by: Priyanka Salunke <priyanka.s.salunke@seagate.com>

### CODE REVIEW

File: s3/replication/common/src/s3replicationcommon/jobs.py [hunk 1]
```diff
@@ -264,8 +264,18 @@ def add_job(self, job):
         self._jobs[job.get_replication_id()] = job
         self._job_id_to_replication_id_map[job.get_job_id()] = \
             job.get_replication_id()
-        # Initial state is queued.
-        self._jobs_queued[job.get_replication_id()] = None
+
+        if self._label == "completed-jobs":
+            # As the state of job is changed to completed, add the job in
```
Comment (rajkumarpatel2602) [inline s3/replication/common/src/s3replicationcommon/jobs.py:269]: if calling object is having label of "completed-jobs" type, then add the job in the jobs_completed set.

File: s3/replication/common/src/s3replicationcommon/jobs.py [hunk 2]
```diff
@@ -264,8 +264,18 @@ def add_job(self, job):
...
+            # jobs_completed
+            self._logger.debug("Jobs[{}]: Adding job with job_id {} into jobs_completed".
+                               format(self._label, job.get_job_id()))
+            self._jobs_completed.add(job.get_replication_id())
+        else:
+            # Initial state is queued.
```
Comment (rajkumarpatel2602) [inline s3/replication/common/src/s3replicationcommon/jobs.py:275]: Initial state is queued for "all-jobs" type object.

File: s3/replication/common/src/s3replicationcommon/jobs.py [hunk 3]
```diff
@@ -320,6 +330,8 @@ def _remove_job(self, replication_id):
                 self._jobs_paused.remove(replication_id)
             else:
                 # COMPLETED/ABORTED/FAILED.
+                self._logger.debug("Jobs[{}]: Removed job with job_id {}.".
```
Comment (rajkumarpatel2602) [inline s3/replication/common/src/s3replicationcommon/jobs.py:333]: You may remove this debug log to maintain the consistency with other cases. We are not printing anything on console while removal of any job.

To optimize above code, 
You may make the get_state() call just once. carry the return in the variable and then use that variable in if else ladder.
e.g.
state = job.get_state()
if state == JobState.INITIAL:
...
elif state == JobState.RUNNINING:
...
...

File: s3/replication/common/src/s3replicationcommon/jobs.py (hunk 1 above)
Comment (priyanka25081999) [reply s3/replication/common/src/s3replicationcommon/jobs.py]: done

File: s3/replication/common/src/s3replicationcommon/jobs.py (hunk 2 above)
Comment (priyanka25081999) [reply s3/replication/common/src/s3replicationcommon/jobs.py]: done

File: s3/replication/common/src/s3replicationcommon/jobs.py (hunk 3 above)
Comment (priyanka25081999) [reply s3/replication/common/src/s3replicationcommon/jobs.py]: done

**Comment by rajkumarpatel2602** [comment]
Looks good.

**Review by kaustubh-d — APPROVED** [review]

**Review by rajkumarpatel2602 — APPROVED** [review]

**Comment by priyanka25081999** [comment]
The code is tested and checked against flake8 and autopep8 and now ready to get merged.

**Review by kaustubh-d — APPROVED** [review]

**Pull request merged**